beancount/query/query_compile_test.py

```diff
@@ -186,6 +186,14 @@ def test_compile_EvalSub(self):
         c_plus = qc.EvalSub(qc.EvalConstant(17), qc.EvalConstant(18))
         self.assertEqual(Decimal, c_plus.dtype)
 
+    def test_compile_EvalAddAmount(self):
+        c_plus = qc.EvalAdd(qc.EvalAmount(Decimal('1.11','USD')), qc.EvalAmount(Decimal('3.55','USD')))
+        self.assertEqual(Decimal, c_plus.dtype)
```

Comment:
Uh!? You testing that the result is of type `Deciamal`. Shouldn't it be of type `Amount`?

Reply:
Yeah, this was a mess with EvalAmount converting to an amount, but this is cleaned up now.